docker-app/qfieldcloud/core/tests/test_jobs.py

```diff
@@ -0,0 +1,129 @@
+    ProcessProjectfileJob,
+    Project,
+)
+from rest_framework import status
+from rest_framework.test import APITransactionTestCase
+
+from .utils import (
+    set_subscription,
+    setup_subscription_plans,
+    testdata_path,
+    wait_for_project_ok_status,
+)
+
+logging.disable(logging.CRITICAL)
+
+VALID_LOCALIZED_LAYER_KEY = "apiary_853c29cb_ead8_4932_a60a_5cba33140b3d"
+INVALID_LAYER_KEY = "apiary_e7d1b542_1b1c_422d_ac75_c14d2b54e472"
+INVALID_LOCALIZED_LAYER_KEY = "area_a7b70aff_23f2_4876_9d4e_fa63036df0b2"
+
+
+class QfcTestCase(APITransactionTestCase):
+    def setUp(self):
+        setup_subscription_plans()
+
+        # Create a user
+        self.u1 = Person.objects.create_user(username="u1", password="abc123")
+        self.t1 = AuthToken.objects.get_or_create(user=self.u1)[0]
+
+        # Create a project
+        self.p1 = Project.objects.create(name="p1", is_public=False, owner=self.u1)
+
+    def test_create_job_bad_layer_handler_extracted_values(self):
+        # Test that BadLayerHandler is parsing data properly during package and process projectfile jobs
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.t1.key)
+
+        # Push the QGIS project and datasource GPKG files
+        for local_filename, upload_filename in [
+            ("bumblebees.gpkg", "bumblebees.gpkg"),
+            ("simple_bumblebees_wrong_localized.qgs", "project.qgs"),
+        ]:
+            file_path = testdata_path(local_filename)
+            response = self.client.post(
+                f"/api/v1/files/{self.p1.id}/{upload_filename}/",
+                {
+                    "file": open(file_path, "rb"),
+                },
+                format="multipart",
+            )
+            self.assertTrue(status.is_success(response.status_code))
+
+        # Create ProcessProjectfileJob and wait for it to finish
+        processprojectfile_job = ProcessProjectfileJob.objects.create(
+            type=Job.Type.PROCESS_PROJECTFILE,
+            project=self.p1,
+            created_by=self.u1,
+        )
+        self.assertEqual(processprojectfile_job.status, Job.Status.PENDING)
+        wait_for_project_ok_status(self.p1)
+
+        processprojectfile_job.refresh_from_db()
+        self.assertEqual(processprojectfile_job.status, Job.Status.FINISHED)
+        self.assertIsNotNone(processprojectfile_job.feedback)
+
+        self.p1.refresh_from_db()
+        self.assertIsNotNone(self.p1.project_details)
+
+        # Create Package job and wait for it to finish
+        set_subscription(
+            self.u1,
+            "keep_10",
+            storage_mb=10,
+            is_premium=True,
+            is_external_db_supported=True,
+        )
+        package_job = PackageJob.objects.create(
+            type=Job.Type.PROCESS_PROJECTFILE,
+            project=self.p1,
+            created_by=self.u1,
+        )
+        self.assertEqual(package_job.status, Job.Status.PENDING)
+        wait_for_project_ok_status(self.p1)
```

Comment:
@suricactus wondering if this test should be split in two : one for the `ProcessProjectfileJob` job, one for the `PackageJob` job :thinking:

Reply:
Yes, better split in two.